src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -29,6 +29,26 @@
 
 logger = _logging.get_az_logger(__name__)
 
+def _resource_client_factory():
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    return get_mgmt_service_client(ResourceManagementClient)
+
+def cf_providers():
+    return _resource_client_factory().providers
+
+def register_providers():
+    providers = cf_providers()
+
+    namespaces = ['Microsoft.Network', 'Microsoft.Compute', 'Microsoft.Storage']
+    for namespace in namespaces:
+        state = providers.get(resource_provider_namespace=namespace)
+        if state.registration_state != 'Registered':
+            logger.info('registering {}', namespace)
+            providers.register(resource_provider_namespace=namespace)
```

Comment:
The issue mentioned at #1309 might not be addressed by this change, because that issue surfaced during deployment, when arm did try to register the provider for you, but failed after 1 minute wait.